fish-rust/src/builtins/functions.rs

```diff
@@ -0,0 +1,413 @@
+                current_func
+            ));
+            builtin_print_error_trailer(parser, streams, cmd);
+            return STATUS_CMD_ERROR;
+        }
+
+        function::set_desc(current_func, opts.description, parser);
+        return STATUS_CMD_OK;
+    }
+
+    if opts.report_metadata {
+        if args.len() != 1 {
+            streams.err.append(wgettext_fmt!(
+                BUILTIN_ERR_ARG_COUNT2,
+                cmd,
+                // This error is
+                // functions: --details: expected 1 arguments; got 2
+                // The "--details" was "argv[optind - 1]" in the C++
+                // which would just give the last option.
+                // This is broken because you could do `functions --details --verbose foo bar`, and it would error about "--verbose".
+                "--details",
+                1,
+                args.len()
+            ));
+            return STATUS_INVALID_ARGS;
+        }
+        if let Some(props) = function::get_props_autoload(args[0], parser) {
+            if let Some(cpf) = &props.copy_definition_file {
+                streams.out.append(sprintf!("%ls\n", cpf));
+            } else if let Some(df) = &props.definition_file {
+                streams.out.append(sprintf!("%ls\n", df));
+            } else {
+                streams.out.append(sprintf!("%ls\n", "stdin"));
+            }
+            if opts.verbose {
+                if props.copy_definition_file.is_some() {
+                    if let Some(df) = &props.definition_file {
+                        streams.out.append(sprintf!("%ls\n", df));
+                    } else {
+                        streams.out.append(sprintf!("%ls\n", "stdin"));
+                    }
+                } else {
+                    streams.out.append(sprintf!(
+                        "%ls\n",
+                        if props.is_autoload.load() {
+                            "autoloaded"
+                        } else {
+                            "n/a"
+                        }
+                    ));
+                }
+                streams.out.append(sprintf!(
+                    "%d\n%ls\n",
+                    props.definition_lineno(),
+                    if props.shadow_scope {
+                        "scope-shadowing"
+                    } else {
+                        "no-scope-shadowing"
+                    }
+                ));
+                if !props.description.is_empty() {
+                    streams.out.append(sprintf!(
+                        "%ls\n",
+                        escape_string(
+                            &props.description,
+                            EscapeStringStyle::Script(
+                                EscapeFlags::NO_PRINTABLES | EscapeFlags::NO_QUOTED
+                            )
+                        ),
+                    ));
+                }
+            }
+        } else if opts.verbose {
+            streams.out.append(L!("n/a\nn/a\n0n/a\nn/a\n"));
```

Comment:
This would need to be adjusted if we add any more metadata - the C++ would set the different entries to "n/a" on initialization.

I could also add mutation like that or maybe return multiple values or something?